src/Shared/SignalVersionHelper.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation;
+
+internal static class SignalVersionHelper
```

Comment:
Curious, why is this called "signal"?

Reply:
In Contrib repo used also for meter versioning

Reply:
`InstrumentationScopeHelper` could be a better name.